packages/filter-lockfile/src/filterLockfileByImportersAndEngine.ts

```diff
@@ -162,7 +162,7 @@ function pkgAllDeps (
         ...(opts.include.optionalDependencies ? pkgSnapshot.optionalDependencies : {}),
       })
       .map(([pkgName, ref]) => {
-        if (pkgSnapshot.peerDependencies?.[pkgName]) return null
+        if (pkgSnapshot.peerDependenciesMeta?.[pkgName]?.optional) return null
```

Comment:
I honestly don't remember why this if statement is needed here. Maybe it should be removed.